airbyte-ci/connectors/pipelines/pipelines/helpers/run_steps.py

```diff
@@ -0,0 +1,233 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+"""The actions package is made to declare reusable pipeline components."""
+
+from __future__ import annotations
+from dataclasses import dataclass, field
+import inspect
+from pipelines import main_logger
+
+from typing import TYPE_CHECKING, Callable, Dict, List, Tuple, Union
+
+import asyncer
+from pipelines.models.steps import StepStatus, StepResult
+
+RESULTS_DICT = Dict[str, StepResult]
+ARGS_TYPE = Union[Dict, Callable[[RESULTS_DICT], Dict]]
+
+if TYPE_CHECKING:
+    from pipelines.models.steps import Step, StepResult
+
+
+@dataclass
+class RunStepOptions:
+    """Options for the run_step function."""
+    fail_fast: bool = True
+    skip_steps: List[str] = field(default_factory=list)
+    log_step_tree: bool = True
+
+
+@dataclass(frozen=True)
+class StepToRun:
```

Comment:
Just opening the discussion here.
Could we rename `StepToRun` to `Step`
And `Step` to `Operation`?

A `Operation.__init__` would take a `Step` instance and would access parameters from `Step.args`.

`Step` instances would be "singleton"
`Operation` instance would not be singleton: an operation can be  re-used in multiple step.
And I think `Operation` might be worth being functions as they'll be stateless.

Reply:
Im all for finding a way to cut down the awkwardness (and layers) of these abstraction.


Hearing works like "singleton", "would be" and "I think" is telling me that this might be better tackled on its own so we can just wrestle with restructuring the taxonomy and not restructuring the taxonomy AND adding new functionality.

WDYT?